sdk/forms/src/main/java/com/klaviyo/forms/bridge/LifecycleObserver.kt

```diff
@@ -3,31 +3,34 @@ package com.klaviyo.forms.bridge
 import com.klaviyo.analytics.Klaviyo
 import com.klaviyo.core.Registry
 import com.klaviyo.core.lifecycle.ActivityEvent
+import com.klaviyo.core.lifecycle.ActivityObserver
 import com.klaviyo.forms.InAppFormsConfig
 import com.klaviyo.forms.bridge.JsBridge.LifecycleEventType.background
 import com.klaviyo.forms.bridge.JsBridge.LifecycleEventType.foreground
 import com.klaviyo.forms.reInitializeInAppForms
 
-internal class LifecycleObserver : JsBridgeObserver {
+internal class LifecycleObserver : JsBridgeObserver, ActivityObserver {
     private var lastBackgrounded: Long? = null
 
     private val sessionTimeoutMs: Long
         get() = Registry.get<InAppFormsConfig>().getSessionTimeoutDuration().inWholeMilliseconds
 
-    override fun startObserver() = Registry.lifecycleMonitor.onActivityEvent(::onLifecycleEvent)
+    override fun startObserver() = Registry.lifecycleMonitor.onActivityEvent(this)
 
     override fun stopObserver() {
         lastBackgrounded = null
-        Registry.lifecycleMonitor.offActivityEvent(::onLifecycleEvent)
+        Registry.lifecycleMonitor.offActivityEvent(this)
     }
 
-    private fun onLifecycleEvent(activity: ActivityEvent) = when (activity) {
-        // App foregrounded
-        is ActivityEvent.FirstStarted -> onForeground()
-        // App backgrounded
-        is ActivityEvent.AllStopped -> onBackground()
-        // Ignore all others
-        else -> Unit
+    override fun invoke(activity: ActivityEvent) {
```

Comment:
Also, I've been cleaning up this old stylistic sin of mine wherever I see it: using expressions should be reserved for methods where we are actually returning something other than `Unit`. Else its just confusing.